kimchi/src/circuits/constraints.rs

```diff
@@ -119,6 +125,10 @@ pub struct ConstraintSystem<F: FftField> {
     #[serde(skip)]
     pub fr_sponge_params: ArithmeticSpongeParams<F>,
 
+    /// Foreign field modulus parameter
+    #[serde_as(as = "Vec<o1_utils::serialization::SerdeAs>")]
+    pub foreign_modulus: Vec<F>,
```

Comment:
Currently unused; remove?

Reply:
It's used.  I set it in the unit tests.

Reply:
Replace with a boolean flag? These gates aren't really doing any foreign field arithmetic, and they don't use this modulus..

Reply:
Which gates do you mean?

I could remove it, but then I will need to put it right back in the next PR because the next constraints will be using it.  My goal for this PR was that it hooks up everything I need for the next set of constraints using the foreign modulus.

The circuit developer needs a way to specify the foreign modulus.  How do you propose this is specified and passed to the gate?  Izaak indicated he didn't want it passed as a gate input nor in the coefficients.  If there's a better or more desirable way, I'll do it.  What's your idea?

Reply:
I see you mean to extract everything that's range check related into a `RangeCheck` gate.  For the range check gate this indeed can be removed.  I'll do this.